### ISSUE TITLE
🐛 Bug Report: CircleCI plugin doesn't support new CircleCI slug format

### ISSUE THREAD

**Issue by rianfowler** [backstage/backstage#21013]
### 📜 Description

When I add an annotation like this to my catalog entry: 

`circleci.com/project-slug: circleci/[some string]/[some string]`

and I have configured the proxy for the CircleCI plugin,

when I visit the ci/cd page for my catalog entry 

Then I get a 404 page. When I inspect the console, it looks like it's hitting bitbucket/[somestring]/[somestring] rather than circleci/[somestring]/[somestring]

### 👍 Expected behavior

When I visit the ci/cd page it should use the new slug format provided by circleCI as documented here: circleci.com/project-slug: circleci/Tvi5amfQm34yJAWpcEh1Ct/7sfMPJ27kf18v48Eh

### 👎 Actual Behavior with Screenshots

just 404s 

### 👟 Reproduction steps

Go through the instructions to set up the CircleCI plugin 

### 📃 Provide the context for the Bug.

I was able to make this work by adding a circleCI case to this switch statement: https://github.com/backstage/backstage/blob/83a519f5a8a18af9f8ad9c66f11f97171c19efb4/plugins/circleci/src/state/useBuilds.ts#L115

so rather than falling back to "bitbucket" when my annotation says "circleci", by it will use the appropriate slug. 

This is what I added locally to make it work: 

```
function mapVcsType(vcs) {
  switch (vcs) {
    case "circleci":
      return "circleci";
    case "gh":
    case "github":
      return GitType.GITHUB;
    default:
      return GitType.BITBUCKET;
  }
}
```

### 👀 Have you spent some time to check if this bug has been raised before?

- [X] I checked and didn't find similar issue

### 🏢 Have you read the Code of Conduct?

- [X] I have read the [Code of Conduct](https://github.com/backstage/backstage/blob/master/CODE_OF_CONDUCT.md)

### Are you willing to submit PR?

None

**Comment by tudi2d** [comment]
Thanks for raising this! Heads-up that the plugin moved to https://github.com/CircleCI-Public/backstage-plugin. We should definitly make it clearer that the ownership changes so I created #21104 fixing existin references.

TL;DR: change `@backstage/plugin-circleci` to `@circleci/backstage-plugin` in your dependencies; if the problem persist please raise this issue on the plugin repository of CircleCI :)
